crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -1190,7 +1190,7 @@ impl OlmMachine {
             }
         };
 
-        debug!("Received a to-device event");
+        debug!(?event, "Received a to-device event");
```

Comment:
This will log the some fields of the event, which are already attached to the span, multiple times. We also avoid logging the full contents of every event we receive, people usually frown upon that, i.e. people might send custom events over to-device which they wouldn't like to end up in their logs.

Here: https://github.com/matrix-org/matrix-rust-sdk/blob/21ca4e98f10f453175e38aba42417c345704ef6f/crates/matrix-sdk-crypto/src/machine.rs#L1133

or here: https://github.com/matrix-org/matrix-rust-sdk/blob/21ca4e98f10f453175e38aba42417c345704ef6f/crates/matrix-sdk-crypto/src/machine.rs#L869-L881

might be a better place to log the content of a withheld event. Are we also sure that the issue wouldn't like some log when a decryption failure happens that contains a withheld code?

Reply:
Yup, fair.

> Are we also sure that the issue wouldn't like some log when a decryption failure happens that contains a withheld code?

I don't know. I'm surprised if that doesn't happen anyway? It's very much not what it asks for, and it feels like scope creep.

Reply:
I've updated the code and description of the PR. PTAL